helpers/build_packages.sh

```diff
@@ -151,7 +151,60 @@ if [ "$BUILDMW" = "1" ]; then
 
     pushd $ANDROID_ROOT/hybris/mw > /dev/null
 
-    if [ "$BUILDMW_REPO" = "" ]; then
+    MANIFEST_LOOKUP=$ANDROID_ROOT
+    while true; do
+        MANIFEST=$MANIFEST_LOOKUP/.repo/manifest.xml
+        if [ -f $MANIFEST -o "MANIFEST_LOOKUP" == "$(dirname $MANIFEST_LOOKUP)" ]; then
+            break
+        fi
+        MANIFEST=""
+        MANIFEST_LOOKUP=$(dirname $MANIFEST_LOOKUP)
+    done
+
+    if [ ! "$BUILDMW_REPO" = "" ]; then
+        # No point in asking when only one mw package is being built
+        BUILDMW_QUIET=1
+        if [ -z "$BUILDSPEC_FILE" ]; then
+            buildmw -u $BUILDMW_REPO || die
+        else
+            # Supply all given spec files from $BUILDSPEC_FILE array prefixed with "-s"
+            buildmw -u $BUILDMW_REPO "${BUILDSPEC_FILE[@]/#/-s }" || die
+        fi
+    elif [ -n $MANIFEST ] &&
+       [ -n "$(grep -l hybris/mw $MANIFEST)" ]; then
```

Comment:
Can you just use the result of grep by doing something like this?:
```
if grep -ql expr $file 
then
..
fi
``` 
This would avoid a sub shell and the need for test.